### ISSUE TITLE
"Show more" button in symbols sidebar flickers with "x symbols total"

### ISSUE THREAD

**Issue by mrnugget** [sourcegraph/sourcegraph#18904]
1. Open a symbols sidebar here: https://sourcegraph.com/github.com/rust-lang/rust/-/tree/compiler/rustc_ast_passes
2. Click on "Show more" at the bottom of the symbols sidebar
3. "x symbols total" shows up and then disappears

Video:

https://user-images.githubusercontent.com/1185253/110104252-33fa8a80-7da7-11eb-9987-1999d7e804e3.mp4

**Comment by mrnugget** [comment]
The same happens with the "repositories" dropdown thing on Sourcegraph.com:


https://user-images.githubusercontent.com/1185253/110104342-542a4980-7da7-11eb-8530-4874a5f93a0d.mp4

**Comment by pdubroy** [comment]
I'll investigate. Might be a good first bug for me to fix.

Thanks @mrnugget for filing this.

**Comment by pdubroy** [comment]
@AlicjaSuska What is the desired behaviour here? I can eliminate the height change, but I'm not sure if this is better:

![sourcegraph-issue-18904](https://user-images.githubusercontent.com/17293/110111882-4c6fa280-7db1-11eb-86f4-86a00bb5d06a.gif)

Maybe we want to show something like "Showing X items - [Show more](#foo)"

**Comment by quinnkeast** [comment]
In @AlicjaSuska's absence, I suggest something along the lines of what you suggest:

<img width="901" alt="CleanShot 2021-03-05 at 13 18 56@2x" src="https://user-images.githubusercontent.com/6304497/110114726-64e1bc00-7db5-11eb-8e80-b6dc8b367017.png">

**Comment by pdubroy** [comment]
I just wanted to follow up here on one of @umpox's comments in Figma:

> if the entire grey row is clickable then the button should semantically wrap the entire row. In that case, screen readers won't read this well and the focus outline won't be on "Show more". I think the majority of our buttons should just read as action verbs to avoid this kind of confusion.

Here is one way we could implement this:

```
<!DOCTYPE html>
<head><title>#18904</title></head>
<style>
    .show-more-info {
        position: relative;
    }
    .details {
        position: absolute;
        pointer-events: none;
    }
    button {
        padding-left: 140px;
    }
</style>

<div class="show-more-info">
    <div class="details">Showing 101 cats</div>
    <button>Show more</button>
</div>
```

And here is how it looks/behaves (obviously we'd want to tweak margins etc.):

![18904-show-more](https://user-images.githubusercontent.com/17293/111457737-dce6a500-8718-11eb-8294-f40ff83aec70.gif)

@umpox - wdyt? Do you think this is an acceptable solution, or is it likely to cause other problems?

**Comment by umpox** [comment]
I think in this case we should just aim to override the screen reader behaviour and let the focus outline wrap the entire grey area. Something like:

```
<button aria-label="Show more">
    <span>Showing 101 cats</span>
    <span>Show more</span>
</div>
```

I'll be honest, I find the approach below less confusing where the button is clearly labelled as an action verb but if that layout is something we don't want then I'm OK just implementing the above code

![image](https://user-images.githubusercontent.com/9516420/111460164-8f1b6e00-8713-11eb-91db-90efb755a137.png)

**Comment by pdubroy** [comment]
@umpox Good point, using aria attributes to customize the screen reader behavior is probably the best option. Either way, it sounds like we have an acceptable technical solution to the design.

**Comment by AlicjaSuska** [comment]
Thank you @umpox and @pdubroy for finding the solution.

Design note: I think having a grey bar at the bottom visually works better with the current table visuals (dividers, a lot of spacing). @quinnkeast 's [proposition](https://www.figma.com/file/kfbnuoYCJD80mYCGJF30kF/Show-more-button-in-symbols-sidebar-flickers-with-x-symbols-total-18904-Final-review?node-id=762%3A0) looks amazing but it would require us to increase the scope of changes. I think we can come back to it during the refresh.

**Comment by pdubroy** [comment]
fyi @AlicjaSuska - I discovered another variant of this component:

![CleanShot 2021-03-23 at 12 01 09](https://user-images.githubusercontent.com/17293/112137042-0f3c4a80-8bd0-11eb-813e-c734c4b004d0.gif)

As seen here: https://sourcegraph.com/github.com/sourcegraph/sourcegraph
